Bottom: 17d9176e66aa6382b512b4e38ca425e5d094dc2d
Top:    bb2f9d882d41256b2f3d46fd206aa7d741ad7b7d
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-12-17 02:16:30 +0000

Refresh of tmp

---

diff --git a/src/rats.c b/src/rats.c
index 29211bc..eab2370 100644
--- a/src/rats.c
+++ b/src/rats.c
@@ -341,6 +341,11 @@ TransferNet (NetListTypePtr Netl, NetTypePtr SourceNet, NetTypePtr DestNet)
 {
   ConnectionTypePtr conn;
 
+  if (SourceNet == NULL) {
+    printf ("FOOBAR\n");
+    return;
+  }
+
   /* It would be worth checking if SourceNet is NULL here to avoid a segfault. Seb James. */
   CONNECTION_LOOP (SourceNet);
   {
@@ -657,10 +662,10 @@ DrawShortestRats (NetListTypePtr Netl, void (*funcp) ())
 		  changed = true;
 		}
 	    }
-
-	  /* copy theSubnet into the current subnet */
-	  TransferNet (Netl, theSubnet, subnet);
 	}
+
+        /* copy theSubnet into the current subnet */
+        TransferNet (Netl, theSubnet, subnet);
     }
 
   /* presently nothing to do with the new subnet */
